src/diffusers/models/resnet.py

```diff
@@ -666,7 +666,193 @@ def forward(self, input_tensor, temb, scale: float = 1.0):
         return output_tensor
 
 
-# unet_rl.py
+class Upsample3D(nn.Module):
+    """A 3D upsampling layer. Reshapes the input tensor to video like tensor, applies upsampling conv,
+    converts it back to the original shape.
+
+    Parameters:
+        channels (`int`):
+            number of channels in the inputs and outputs.
+        out_channels (`int`, optional):
+            number of output channels. Defaults to `channels`.
+    """
+
+    def __init__(self, channels, out_channels=None):
+        super().__init__()
+        self.channels = channels
+        self.out_channels = out_channels or channels
+
+        self.conv = nn.Conv2d(self.channels, self.out_channels, 3, padding=1)
+
+    def forward(self, hidden_states, output_size=None):
+        assert hidden_states.shape[1] == self.channels
```

Comment:
```suggestion
        if hidden_states.shape[1] != self.channels:
            raise ValueError("...") 
```

Can we add a nice error message here?

Reply:
Done